From 81bbae037555624f1c76ad77b8d8db70d50ace81 Mon Sep 17 00:00:00 2001
From: Andy Green <andy.green@linaro.org>
Date: Fri, 1 Feb 2013 11:17:02 +0800
Subject: [PATCH 04/18] remove one more mention of broadcast callback

Signed-off-by: Andy Green <andy.green@linaro.org>
---
 lib/libwebsockets.h        |    5 -----
 libwebsockets-api-doc.html |    6 ------
 2 files changed, 11 deletions(-)

--- a/lib/libwebsockets.h
+++ b/lib/libwebsockets.h
@@ -133,7 +133,6 @@
 	LWS_CALLBACK_SERVER_WRITEABLE,
 	LWS_CALLBACK_HTTP,
 	LWS_CALLBACK_HTTP_FILE_COMPLETION,
-	LWS_CALLBACK_BROADCAST,
 	LWS_CALLBACK_FILTER_NETWORK_CONNECTION,
 	LWS_CALLBACK_FILTER_PROTOCOL_CONNECTION,
 	LWS_CALLBACK_OPENSSL_LOAD_EXTRA_CLIENT_VERIFY_CERTS,
@@ -389,10 +388,6 @@
  *
  *	LWS_CALLBACK_CLOSED: when the websocket session ends
  *
- *	LWS_CALLBACK_BROADCAST: signal to send to client (you would use
- *				libwebsocket_write() taking care about the
- *				special buffer requirements
- *
  *	LWS_CALLBACK_RECEIVE: data has appeared for this server endpoint from a
  *				remote client, it can be found at *in and is
  *				len bytes long
--- a/libwebsockets-api-doc.html
+++ b/libwebsockets-api-doc.html
@@ -596,12 +596,6 @@
 <blockquote>
 when the websocket session ends
 </blockquote>
-<h3>LWS_CALLBACK_BROADCAST</h3>
-<blockquote>
-signal to send to client (you would use
-<b>libwebsocket_write</b> taking care about the
-special buffer requirements
-</blockquote>
 <h3>LWS_CALLBACK_RECEIVE</h3>
 <blockquote>
 data has appeared for this server endpoint from a
