Description: Fix a spelling error (connnection).
Author: Peter Pentchev <roam@ringlet.net>
Forwarded: not yet
Last-Update: 2013-02-06

--- a/test-server/test-ping.c
+++ b/test-server/test-ping.c
@@ -428,7 +428,7 @@
 				 "origin", protocols[PROTOCOL_LWS_MIRROR].name,
 								  ietf_version);
 		if (ping_wsi[n] == NULL) {
-			fprintf(stderr, "client connnection %d failed to "
+			fprintf(stderr, "client connection %d failed to "
 								"connect\n", n);
 			return 1;
 		}
